xdsl/dialects/onnx.py

```diff
@@ -673,6 +675,94 @@ def verify_(self) -> None:
+        elif self.value_ints:
+            # In the ONNX-MLIR format, integer types are abstracted, so only the integer data is extracted
+            # for printing purposes.
+            arr: list[int] = [value.value.data for value in self.value_ints]
+            return arr, "value_ints"
+        else:
+            # TODO: value_string, value_strings, value_floats
+            raise NotImplementedError()
+
+    def print(self, printer: Printer):
+        if self.value is not None:
+            printer.print_string(" ")
+            printer.print_attribute(self.value)
+        printer.print_string(" ")
+        printer.print_string("{")
+        try:
+            value, value_type = self.get_value()
+        except NotImplementedError:
+            # handle the case where get_value() is not implemented
+            value, value_type = None, None
+        if value and value_type:
+            if value_type == "value_ints":
+                printer.print_string(value_type)
+                printer.print_string(" = ")
+                printer.print(value)
+            else:
+                printer.print_string(value_type)
+                printer.print_string(" = ")
+                printer.print_attribute(value)
+        printer.print_string("}")
+        printer.print_string(" : ")
+        printer.print_attribute(self.output.type)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        value = None
+        value_float = None
+        value_floats = None
+        value_int = None
+        value_ints = None
+        value_string = None
+        value_strings = None
+        output_type = None
+        if parser.parse_optional_punctuation("{"):
+            if parser.parse_optional_keyword("value_float"):
+                parser.parse_punctuation("=")
+                value_float = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_floats"):
+                parser.parse_punctuation("=")
+                value_floats = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_int"):
+                parser.parse_punctuation("=")
+                value_int = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_ints"):
+                parser.parse_punctuation("=")
+                value_ints = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_string"):
+                parser.parse_punctuation("=")
+                value_string = parser.parse_attribute()
+            elif parser.parse_optional_keyword("value_strings"):
+                parser.parse_punctuation("=")
+                value_strings = parser.parse_attribute()
+            parser.parse_punctuation("}")
+            parser.parse_punctuation(":")
+            output_type = parser.parse_type()
+        else:
+            value = parser.parse_attribute()
+            if isinstance(value, DenseIntOrFPElementsAttr):
+                output_type = value.type
```

Comment:
similarly here, we can just parse an attribute. If it's a `dense`, then we continue to parse an optional dict. If it's not `dense`, then it must be an extra attributes dict. I'd expect something like this to work.